torchao/ops.py

```
@@ -70,6 +70,9 @@
 lib.define(
     "da8w4_linear_cpu(Tensor input, Tensor input_scales, Tensor input_qzeros, Tensor weight, Tensor weight_scales, Tensor weight_qzeros, Tensor compensation, Tensor? bias, ScalarType output_dtype) -> Tensor"
 )
+lib.define(
+    "qembeddingbag(Tensor qweight, Tensor indices, Tensor offsets, Tensor weight_scale, float o_scale, int mode, bool include_last_offset) -> Tensor"
```

is this the same as https://github.com/pytorch/pytorch/blob/371eacb2ae4ecdabc52ea4634ed21558df2f3bab/aten/src/ATen/native/native_functions.yaml#L2368C1-L2369C1? with the only difference of qweight being float8?

---

@jerryzh168 Thanks for reviewing. Yes, I think so, except that the implementation in this PR has limited functionality so far.

---

I think we should add this to pytorch directly if that's the case, float8 is a native dtype in pytorch, so I think it makes most of the sense to just add the functionality there, we can error out if some arg combination is not supported or invalid for float8

---

Intel's platform has fp8 instructions. When we are ready, we hope to update this kernel based on fp8 instructions. As far as I know, the latest GCC is required. Is it difficult to support in PyTorch?

---

I'm not sure, can you open an issue for this in pytorch/pytorch?

---

And for PT API, PT use qtensor and no need to use scale, and we use scale here. Could we land on ao?

---

oh I meant can we add to normal embedding op in PT: https://github.com/pytorch/pytorch/blob/f5bf5147ad18994c9a6e0f565d7831362bf5a18a/aten/src/ATen/native/native_functions.yaml#L2360, not quantized one

---

So, since this PR adds a quantized version of this op, do you think it better to be added in Torchao rather than in torch core? Thanks.

---

OK, sounds good, maybe rename this to `scaled_embedding_bag` to follow these ops: https://github.com/pytorch/pytorch/blob/31a41daff49f2cde941d8b9e35cb2eaeeb606c0d/aten/src/ATen/native/native_functions.yaml#L7135

---

Done